apps/koda/src/comments/comments.controller.ts

```diff
@@ -0,0 +1,81 @@
+import { CommentsService } from "./comments.service";
+import {
+  Controller,
+  Post,
+  Body,
+  Param,
+  Request,
+  UseGuards,
+  Get,
+  Delete,
+  Patch,
+  ForbiddenException,
+} from '@nestjs/common';
+import {
+  ApiTags,
+  ApiOperation,
+  ApiResponse,
+} from '@nestjs/swagger';
+
+import { JwtAuthGuard, type UUID, type AuthenticatedRequest } from '@fubs/shared';
+import { CreateCommentDto } from './dto/create-comment.dto';
+import { UpdateCommentDto } from "./dto/update-comment.dto";
+import { WorkspaceMemberGuard } from '../auth/guards/workspace-member.guard';
+
+@ApiTags('comments')
+@Controller('comments')
+export class CommentsController {
+  constructor(private readonly commentsService: CommentsService) {}
+
+  @Post()
+  @UseGuards(JwtAuthGuard, WorkspaceMemberGuard)
+  @ApiOperation({ summary: 'Create a comment' })
+  @ApiResponse({ status: 201, description: 'Comment created successfully' })
+  createComment(@Body() createCommentDto: CreateCommentDto) {
+    return this.commentsService.createComment(createCommentDto);
+  }
+
+  @Get(':taskId')
+  @UseGuards(JwtAuthGuard, WorkspaceMemberGuard)
+  @ApiOperation({ summary: 'Get comments by task ID' })
+  @ApiResponse({ status: 200, description: 'Comments retrieved successfully' })
+  getCommentsByTaskId(@Param('taskId') taskId: UUID) {
+    return this.commentsService.getCommentsByTaskId(taskId);
+  }
+
+  @Patch(':commentId')
+  @UseGuards(JwtAuthGuard, WorkspaceMemberGuard)
+  @ApiOperation({ summary: 'Update a comment' })
+  @ApiResponse({ status: 200, description: 'Comment updated successfully' })
+  async updateComment(
+    @Param('commentId') commentId: UUID,
+    @Body() updateCommentDto: UpdateCommentDto,
+    @Request() req: AuthenticatedRequest
+  ) {
+    const userId = req.user.id;
+    const comment = await this.commentsService.getCommentById(commentId);
+
+    if (comment.createdBy !== userId) {
+      throw new ForbiddenException('You are not allowed to update this comment');
+    }
+
+    // Add commentId to the DTO for the service
+    const updateDto = { ...updateCommentDto, commentId };
+    return this.commentsService.updateComment(updateDto);
+  }
+
+  @Delete(':commentId')
+  @UseGuards(JwtAuthGuard, WorkspaceMemberGuard)
```

Comment:
suggestion: As it uses the same guards for all verbs, you can add it to the @Controller('comments')

Reply:
[1530c82](https://github.com/tassioFront/fubs/pull/43/commits/1530c824ddf919fef96161f843e98bf3a7881b28)